authentication/LaravelInertia.js

```diff
@@ -0,0 +1,148 @@
+/*
+ * A script to provide authentication for Laravel InertiaJS apps.
+ *
+ * First it makes a GET request and obtains the XSRF-TOKEN and Cookie Session from the response body.
+ *
+ * Then it makes a POST request with a body which contains username, password and X-XSRF-TOKEN.
+ *
+ * A successful login will result in a 302 redirect. If this happens, a GET request is made to the redirect URL.
+ *
+ * Every request made by this script is logged separately to the History tab.
+ */
+
+
+function authenticate(helper, paramsValues, credentials) {
+
+  var AuthenticationHelper = Java.type('org.zaproxy.zap.authentication.AuthenticationHelper');
+  var HttpRequestHeader = Java.type("org.parosproxy.paros.network.HttpRequestHeader");
+  var HttpHeader = Java.type("org.parosproxy.paros.network.HttpHeader");
+  var URI = Java.type("org.apache.commons.httpclient.URI");
+
+  var targetURL = paramsValues.get("Target URL");
+  var baseURL = targetURL.match(/^(.+?[^\/:](?=[?\/]|$))/i)[1];
+
+  //
+  // First, make a GET request to the login page to get and extract the
+  // csrfmiddlewaretoken from it.
+  //
+
+  // Build message.
+  var firstRequestURI = new URI(targetURL, false);
```

Comment:
I'd have expected the URLs to be properly encoded.

Reply:
I don't understand the problem, it's how I saw it being done here

https://github.com/GoldraK/community-scripts/blob/main/authentication/DjangoAuthentication.js#L29